controllers/openstackdataplanenode_controller.go

```diff
@@ -234,17 +233,60 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 		if result.RequeueAfter > 0 {
 			return result, nil
 		}
-	}
 
-	r.Log.Info("Set DataPlaneNodeReadyCondition true")
-	instance.Status.Conditions.Set(condition.TrueCondition(dataplanev1beta1.DataPlaneNodeReadyCondition, dataplanev1beta1.DataPlaneNodeReadyMessage))
+		var novaExternalCompute *novav1beta1.NovaExternalCompute
+		result, novaExternalCompute, err = deployment.DeployNovaExternalCompute(
+			ctx,
+			helper,
+			instance,
+			ansibleSSHPrivateKeySecret,
+			inventoryConfigMap,
+			&instance.Status,
+			instance.Spec.NetworkAttachments,
+			instance.Spec.OpenStackAnsibleEERunnerImage)
+		if err != nil {
+			return result, err
+		}
+
+		novaReadyCondition := novaExternalCompute.Status.Conditions.Get(condition.ReadyCondition)
+		if novaReadyCondition != nil {
+			r.Log.Info(fmt.Sprintf("NovaExternalCompute ReadyCondition status: %s", novaReadyCondition.Status))
+		} else {
+			r.Log.Info("NovaExternalCompute ReadyCondition not yet set")
+		}
+
+		mirroredCondition := novaExternalCompute.Status.Conditions.Mirror(dataplanev1beta1.NovaComputeReadyCondition)
+		if mirroredCondition != nil {
+			instance.Status.Conditions.Set(mirroredCondition)
+		}
+
+		if condition.IsError(instance.Status.Conditions.Get(dataplanev1beta1.NovaComputeReadyCondition)) {
```

Comment:
@gibi Are these checks expect the result of the set mirroredCondition to be in place yet?
Just to clarify, there should be no such expectations due to eventual consistency.
If these check work as expected with not yet updated older status conditions, then it is totally fine though.

Reply:
please disregard, I believe I got it now...